### PR TITLE
feat(desktop): render Markdown in captions and History

### PR DESCRIPTION
**Pull request by dastratakos**
## Summary

Luke's replies arrive as Markdown, and the desktop drew the marks as literal characters while the iOS app drew the structure. This brings the desktop up to iOS:

- **`MarkdownMessage`** (renderer): a thin wrapper over `react-markdown` with `remark-gfm` (tables, strikethrough, task lists; single tildes disabled so `~/` paths never strike). Used by the History bubbles, with the timestamp on a line of its own under the words, and by the voice caption under the housing. Sized in `em` so one stylesheet serves the caption's 10px and the thread's 12.5px, with every tint mixed from `currentColor` so the same rules read on the grey bubble and the accent one.
- The component overrides only what this surface says differently: links, headings, and images. Everything else, task checkboxes and tables included, is the library's default, styled in CSS. Line breaks follow GitHub's reading: a blank line separates paragraphs, a lone newline joins lines.

## Boundaries kept

- **Links are drawn, not controls.** `urlTransform` keeps only HTTP(S) destinations; anything else arrives as words. The renderer refuses every navigation and has no bridge door for opening an arbitrary address, so a link names its destination on hover and nothing more. Making a model's link pressable would be a new capability, so it is left as a product decision.
- **Raw HTML is text**, never markup: no `rehype-raw`, and the library escapes it.
- **Headings are styled, not announced**, so a reply does not enter the panel's document outline for a screen reader.
- **Copy still copies the line's own words**, marks included, never model context.
- History stays inside the `ph-no-capture` subtree; nothing new leaves the machine. No setting, act, or guide entry changes.

## The bug the first test run found

Before this PR, every way into History collapsed all whitespace to a single space, so a Markdown reply reached the panel as one run-on line: headings ran into paragraphs and a fenced block became an inline span. The flattening guarded one thing, the line-oriented context item the model reads, where a pasted paragraph must not open a new line, so `conversationHistoryText` now flattens its own copy and the stored words keep their line structure (line endings made uniform, ends trimmed). A stored line is read back under the same normalization. Tests cover the round trip from append to the rendered bubble and the model render's flattening.

## Dependencies

`react-markdown` and `remark-gfm`, pinned, in the desktop renderer only. An earlier revision of this PR hand-rolled the parser in a workspace package; Bugbot found a real strikethrough edge case in it on the first review, which is the kind of maintenance the library absorbs, so it was replaced.

## Verification

- `./scripts/check.sh` passes (repository checks, lint, typecheck, tests including 9 component tests, build).
- Visual evidence: I rendered the History bubbles (you / Luke / event) and the caption with the built `styles.css` in headless Chrome and inspected the capture: heading, nested code in a list item, strikethrough, table alignment, task boxes, quote bar, and the right-aligned stamp all draw correctly. This ran in a Linux sandbox, so `./scripts/verify.sh` was **not** run here and **no physical-notch check was performed**; CI's macOS job runs it. The capture is not attached because the sandbox cannot reach GitHub's editor.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### Automated visual evidence

[Download the deterministic macOS evidence](https://github.com/ReviewStage/luke/actions/runs/34167820246/artifacts/10034804917) · [workflow run](https://github.com/ReviewStage/luke/actions/runs/34167820246)

- Commit: `270383dfe6898c6a55699ec2bc79ead0b561b943`
- Scenario: `smoke`
- Physical-notch check: not performed by CI

### CODE REVIEW

File: packages/markdown/src/markdown.ts [hunk 1]
```diff
@@ -0,0 +1,966 @@
...
+      let canClose = rightFlanking;
+      if (character === "_") {
+        canOpen = leftFlanking && (!rightFlanking || isPunctuation(before));
+        canClose = rightFlanking && (!leftFlanking || isPunctuation(after));
+      }
+      if (character === "~" && run.length > 2) {
+        buffer += run;
+        index += run.length;
+        continue;
+      }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/markdown/src/markdown.ts]: Fixed in the follow-up commit: a tilde run now strikes only when it is exactly two characters, matching Foundation. Test added for two `~/` paths in one sentence.
